front/lib/resources/labs_transcripts_resource.ts

```diff
@@ -0,0 +1,283 @@
+      : null;
+  }
+
+  async setAgentConfigurationId({
+    agentConfigurationId,
+  }: {
+    agentConfigurationId: string | null;
+  }): Promise<
+    Result<
+      void,
+      | {
+          type: "not_found";
+        }
+      | Error
+    >
+  > {
+    if (this.agentConfigurationId === agentConfigurationId) {
+      return new Ok(undefined);
+    }
+
+    try {
+      await LabsTranscriptsConfigurationModel.update(
+        { agentConfigurationId },
+        {
+          where: {
+            id: this.id,
+          },
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async setEmailToNotify({
+    emailToNotify,
+  }: {
+    emailToNotify: string | null;
+  }): Promise<
+    Result<
+      void,
+      | {
+          type: "not_found";
+        }
+      | Error
+    >
+  > {
+    if (this.emailToNotify === emailToNotify) {
+      return new Ok(undefined);
+    }
+
+    try {
+      await LabsTranscriptsConfigurationModel.update(
+        { emailToNotify },
+        {
+          where: {
+            id: this.id,
+          },
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  static async getIsActive({
+    userId,
+    provider,
+  }: {
+    userId: ModelId;
+    provider: LabsTranscriptsProviderType;
+  }): Promise<boolean> {
+    const configuration = await this.findByUserIdAndProvider({
+      userId,
+      provider,
+    });
+    if (!configuration) {
+      return false;
+    }
+    return configuration.isActive;
+  }
+
+  async setIsActive({ isActive }: { isActive: boolean }): Promise<
+    Result<
+      void,
+      | {
+          type: "not_found";
+        }
+      | Error
+    >
+  > {
+    if (this.isActive === isActive) {
+      return new Ok(undefined);
+    }
+
+    try {
+      await LabsTranscriptsConfigurationModel.update(
+        { isActive },
+        {
+          where: {
+            id: this.id,
+          },
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async delete(transaction?: Transaction): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+}
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface LabsTranscriptsHistoryResource
+  extends ReadonlyAttributesType<LabsTranscriptsHistoryModel> {}
+export class LabsTranscriptsHistoryResource extends BaseResource<LabsTranscriptsHistoryModel> {
```

Comment:
No need to export.

Reply:
If I remove I get `'LabsTranscriptsHistoryResource' is declared but never used`